OMEdit/OMEditLIB/Annotations/TextAnnotation.cpp

```diff
@@ -420,34 +421,42 @@ void TextAnnotation::drawAnnotation(QPainter *painter)
   // if absolute font size is defined and is greater than 0 then we don't need to calculate the font size.
   if (mFontSize <= 0) {
     QFontMetrics fontMetrics(painter->font());
-    QRect fontBoundRect = fontMetrics.boundingRect(absMappedBoundingRect.toRect(), Qt::TextDontClip, textString);
-    const qreal xFactor = absMappedBoundingRect.width() / fontBoundRect.width();
-    const qreal yFactor = absMappedBoundingRect.height() / fontBoundRect.height();
+    QRect fontBoundRect = fontMetrics.boundingRect(mappedBoundingRect.toRect(), Qt::TextDontClip, textString);
+    const qreal xFactor = mappedBoundingRect.width() / fontBoundRect.width();
+    const qreal yFactor = mappedBoundingRect.height() / fontBoundRect.height();
     /* Ticket:4256
      * Text aspect when x1=x2 i.e, width is 0.
      * Use height.
      */
-    const qreal factor = (absMappedBoundingRect.width() != 0 && xFactor < yFactor) ? xFactor : yFactor;
+    const qreal factor = (mappedBoundingRect.width() != 0 && xFactor < yFactor) ? xFactor : yFactor;
     qreal fontSizeFactor = font.pointSizeF() * factor;
     // Yes we don't go below Helper::minimumTextFontSize font pt.
     font.setPointSizeF(qMax(fontSizeFactor, Helper::minimumTextFontSize));
     painter->setFont(font);
   }
   /* Try to get the elided text if calculated font size <= Helper::minimumTextFontSize
-   * OR if font size is absolute and text is not multiline.
+   * OR if font size is absolute.
    */
   QString textToDraw = textString;
-  if (absMappedBoundingRect.width() > 1 && ((mFontSize <= 0 && painter->font().pointSizeF() <= Helper::minimumTextFontSize) || (mFontSize > 0 && !Utilities::isMultiline(textString)))) {
+  if (mappedBoundingRect.width() > 1 && ((mFontSize <= 0 && painter->font().pointSizeF() <= Helper::minimumTextFontSize) || mFontSize > 0)) {
```

Comment:
Please put the comment back and check for multiline. It is not possible to get the elided text for multiline strings.